tracer/test/Datadog.Trace.Tests/Debugger/DebuggerExpressionLanguageTests.cs

```diff
@@ -0,0 +1,99 @@
+// <copyright file="DebuggerExpressionLanguageTests.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using AgileObjects.ReadableExpressions;
+using Datadog.Trace.Debugger.Conditions;
+using VerifyTests;
+using VerifyXunit;
+using Xunit;
+
+namespace Datadog.Trace.Tests.Debugger
+{
+    [UsesVerify]
+    public class DebuggerExpressionLanguageTests
+    {
+        public DebuggerExpressionLanguageTests()
+        {
+            Test = new TestStruct()
+            {
+                Collection = new List<string> { "hello", "1st Item", "2nd item", "3rd item" },
+                IntNumber = 42,
+                DoubleNumber = 3.14159,
+                String = "Hello world!",
+                Nested = new TestStruct.NestedObject() { NestedString = "Hello from nested object" }
+            };
+        }
+
+        internal TestStruct Test { get; set; }
+
+        public static IEnumerable<object[]> DSLExamples()
+        {
+            var path = Path.Combine(Directory.GetParent(Environment.CurrentDirectory).Parent.Parent.FullName, "Debugger", "Resources");
+            return Directory.EnumerateFiles(path).Where(f => !Path.GetFileName(f).StartsWith("Skip")).Select(file => new object[] { file });
+        }
+
+        [Theory]
+        [MemberData(nameof(DSLExamples))]
+        public async Task TestCondition(string conditionTestFilePath)
+        {
+            var conditionTest = File.ReadAllText(conditionTestFilePath);
+            var probeCondition = new ProbeCondition(" ", MethodPhase.End, conditionTest);
```

Comment:
Consider refactoring to a static factory method called `ProbeCondition.FromJson` or `ProbeCondition.ParseFromJson`. It feels a bit too implicit that the constructor is responsible for doing all the parsing.